ledger/ledger-api-bench-tool/src/main/scala/com/daml/ledger/api/benchtool/Benchmark.scala

```diff
@@ -28,7 +27,10 @@ object Benchmark {
       reportingPeriod: FiniteDuration,
       apiServices: LedgerApiServices,
       metricsReporter: MetricsReporter,
-  )(implicit ec: ExecutionContext, resourceContext: ResourceContext): Future[Unit] = {
+  )(implicit
+      ec: ExecutionContext,
+      resourceContext: ResourceContext,
+  ): Future[Either[String, Unit]] = {
```

Comment:
Changed this to differentiate between benchmark failure (like too slow stream) and an application failure.

Reply:
The differentiation is introduced not to log errors in the same form as exceptions (with the stack trace, etc.). I could introduce a specific exception class for errors and pattern match but the approach I proposed seems more like "FP".
@nmarton-da would you like me to change this in a follow-up PR?